crm/crm/doctype/appointment/appointment.py

```diff
@@ -44,13 +44,31 @@ def onload(self):
 		self.set_scheduled_reminder_onload()
 
 	def validate(self):
+		self.validate_duplicate_appointment()
 		self.set_missing_values()
 		self.validate_previous_appointment()
 		self.validate_timeslot_validity()
 		self.validate_timeslot_availability()
 		self.clean_remarks()
 		self.set_status()
 
+	def validate_duplicate_appointment(self):
+		existing_appointment = frappe.db.get_value(
```

Comment:
This feature should be configurable instead of enabled by default. Add a checkbox in Appointment Booking Settings or Appointment Type to enable/disable this feature